vmm/src/interrupt.rs

```diff
@@ -190,8 +190,8 @@ impl InterruptSourceGroup for MsiInterruptGroup<IrqRoutingEntry> {
                     format!("mask: No existing route for interrupt index {}", index),
                 ));
             }
-            self.set_gsi_routes(&routes)?;
-            return route.disable(&self.vm);
+            route.disable(&self.vm)?;
+            return self.set_gsi_routes(&routes);
```

Comment:
Ok so you're saying `KVM_IRQFD` must be invoked before `KVM_SET_GSI_ROUTING`. Could you please update the `unmask()` function as well?

Reply:
unmask() set entry.masked false firstly and  irq_rt->map will not affect in KVM_SET_GSI_ROUTING, so it's ok for unmask().
But it should do no harm if we do so, I will update this patch.